File: Documentation/3.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Commands/Attachments/HowTo/Head.cs
using System.Collections.Generic;

using Raven.Abstractions.Data;
using Raven.Client.Document;

namespace Raven.Documentation.Samples.ClientApi.Commands.Attachments.HowTo
{
	public class Head
	{
		private interface IFoo
		{
			#region head_1_0
			Attachment HeadAttachment(string key);
			#endregion

			#region head_2_0
			IEnumerable<Attachment> GetAttachmentHeadersStartingWith(
				string idPrefix,
				int start,
				int pageSize);
			#endregion
		}

		public Head()
		{
			using (var store = new DocumentStore())
			{
				#region head_1_1
				var attachment = store
					.DatabaseCommands
					.HeadAttachment("albums/holidays/sea.jpg"); // null if does not exist
				#endregion
			}

			using (var store = new DocumentStore())
			{
				#region head_2_1
				var attachments = store
					.DatabaseCommands
					.GetAttachmentHeadersStartingWith("albums/holidays/", 0, 10);
				#endregion
			}
		}
	}
}

File: Documentation/3.0/Samples/java/src/test/java/net/ravendb/clientapi/commands/attachments/howto/Head.java
package net.ravendb.clientapi.commands.attachments.howto;

import java.util.List;

import net.ravendb.abstractions.data.Attachment;
import net.ravendb.client.IDocumentStore;
import net.ravendb.client.document.DocumentStore;

@SuppressWarnings("deprecation")
public class Head {

  @SuppressWarnings("unused")
  private interface IFoo {
    //region head_1_0
    @Deprecated
    public Attachment headAttachment(String key);
    //endregion

    //region head_2_0
    @Deprecated
    public List<Attachment> getAttachmentHeadersStartingWith(String idPrefix, int start, int pageSize);
    //endregion
  }

  @SuppressWarnings("unused")
  public Head() throws Exception {
    try (IDocumentStore store = new DocumentStore()) {
      //region head_1_1
      Attachment attachment = store.getDatabaseCommands().headAttachment("albums/holidays/sea.jpg"); // null if does not exist
      //endregion
    }

    try (IDocumentStore store = new DocumentStore()) {
      //region head_2_1
      List<Attachment> attachments = store.getDatabaseCommands().getAttachmentHeadersStartingWith("albums/holidays/", 0, 10);
      //endregion
    }
  }
}
